src/workloads/query/TimeSeriesCoinbaseBinanceQuery.yml

```diff
@@ -0,0 +1,180 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This test exercises the behavior of the Coinbase/Binance query. The query has $addFields as the
+  first stage, followed by a dependent $match, multiple group stages dependent on computed fields,
+  and a sort stage at the end of the pipeline.
+
+Keywords:
+- timeseries
+- aggregate
+- sbe
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  DocumentCount: &DocumentCount 1e6
+  NumGroups: &NumGroups 100
+  Repeat: &Repeat 10
+  Threads: &Threads 1
+  BatchSize: &BatchSize 30000
+  MaxPhases: &MaxPhases 5
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: *Threads
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: CreateTimeSeriesCollection
+  Type: RunCommand
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Operation:
+          OperationMetricsName: CreateTimeSeriesCollection
+          OperationName: RunCommand
+          OperationCommand:
+            {create: *Collection, timeseries: {timeField: "time", metaField: "symbol", granularity: "seconds"}}
+
+- Name: DropAllIndexesUponCreation
+  Type: RunCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Operation:
+          OperationMetricsName: DropAllIndexesUponCreation
+          OperationName: RunCommand
+          OperationCommand:
+            dropIndexes: *Collection
+            index: "*"
+
+- Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        Collection: *Collection
+        DocumentCount: *DocumentCount
+        BatchSize: *BatchSize
+        Document:
+          time: {^IncDate: {start: {^Now: {}}, step: 1000}}
```

Comment:
The first match tries to match all documents within 24 hours from NOW. I selected this date to maximize the number of matches. Do we want to hit a particular % of total documents to match to exercise this workload thoroughly?